### PR TITLE
Assert in any order

### PR DESCRIPTION
**Pull request by aaron-fishman-achillestx**
This PR adds functionality to assert two lists in any order. This is useful for comparing channels that contain complex data structures (json / maps / tuples etc) in an order agnostic manner. It provides a alternative to sorted channels and can accommodate complex data structures against various assertions:
- new GlobalMethod `assertInAnyOrder(List<Object>, List<Object>)`
- 7 examples in the documentation for different use-cases / data structures
- Unit Tests on groovy scripts that ensure `assertInAnyOrder` behaves as expected 
- Updated ChannelOutput to produce ANSI yellow warnings

Does it fix?  https://github.com/askimed/nf-test/issues/36

### CODE REVIEW

File: docs/assertions/assertions.md [hunk 1]
```diff
@@ -30,6 +30,167 @@ with(process.out.imputed_plink2) {
 }
 ```
 
+## Comparing Channels with Using `assertInAnyOrder`
```
Comment (lukfor) [inline docs/assertions/assertions.md:33]: Thanks for all the examples and documentation! :+1:

File: src/main/java/com/askimed/nf/test/lang/extensions/GlobalMethods.java [hunk 2]
```diff
@@ -38,7 +45,18 @@ public static void assertAll(Closure... closures) throws Exception {
 		}
 
 		if (failed > 0) {
-			throw new Exception(Integer.toString(failed) + " of " + Integer.toString(closures.length) + " assertions failed");
+			throw new PowerAssertionError(Integer.toString(failed) + " of " + Integer.toString(closures.length) + " assertions failed");
+		}
+	}
+	
+	public static void assertContainsInAnyOrder(List<Object> list, List<Object> expected) throws Exception {
```
Comment (lukfor) [inline src/main/java/com/askimed/nf/test/lang/extensions/GlobalMethods.java:52]: in documentation `assertInAnyOrder` vs. `assertContainsInAnyOrder`.  I would prefer `assertInAnyOrder`. What do you think?

**Review by lukfor — COMMENTED** [review]
Thanks for the PR. I just have one point about the method name:
- In Documentation: `assertInAnyOrder`
- In Code: `assertContainsInAnyOrder`

I would prefer `assertInAnyOrder`. What do you think?

File: docs/assertions/assertions.md (hunk 1 above)
Comment (aaron-fishman-achillestx) [reply docs/assertions/assertions.md:33]: For sure! Feels like the solution in the end of the day was realising how to better leverage groovy =]

File: src/main/java/com/askimed/nf/test/lang/extensions/GlobalMethods.java (hunk 2 above)
Comment (aaron-fishman-achillestx) [reply src/main/java/com/askimed/nf/test/lang/extensions/GlobalMethods.java:52]: Happy with either really. I do think `assertInAnyOrder` sounds better, but really it's binding for Hamcrest's method, so I'm inclined to be consistent with that. 

@byb121  what do you think?

Comment (byb121) [reply src/main/java/com/askimed/nf/test/lang/extensions/GlobalMethods.java:52]: It seems to me that Hamcrest already gained popularity with developers judging by the repo's stars. Although I like the shorter form as well, I think the longer form could be easier to guess its function for many devs.

Comment (lukfor) [reply src/main/java/com/askimed/nf/test/lang/extensions/GlobalMethods.java:52]: Good arguments, let us go with the longer form to be consistent with Hamcrest.

Comment (lukfor) [reply src/main/java/com/askimed/nf/test/lang/extensions/GlobalMethods.java:52]: Awesome! Thanks. If everything is fine from your side, I will merge the pull request

Comment (aaron-fishman-achillestx) [reply src/main/java/com/askimed/nf/test/lang/extensions/GlobalMethods.java:52]: Hey @lukfor  yeah all good for me!

**Pull request merged**